elastic/kibana
# Remove securityOss plugin

Resolves #104152.

Note: we don't ship the OSS distribution anymore, but the security plugin can be disabled in 7.16 (it won't be disable-able in 8.0+).
If someone is using the Default distribution of 7.16 _and_ they have the security plugin disabled, then they won't see the Security Checkup message when they open Kibana for the first time. This edge case is an acceptable tradeoff to eliminating an entire plugin that we don't need anymore.

This PR also deprecates the `security.showInsecureClusterWarning: false` setting, renaming it to `xpack.security.showInsecureClusterWarning: false`. This setting does not appear in our documentation.

## Testing

Existing functionality was refactored, here are the steps to test and make sure it still works as expected.

### 1. Insecure cluster toast

Kibana shows the toast when all three of these conditions are true:
* The warning is enabled in kibana.yml (default),
* Security is disabled, and
* Kibana has some data in the cluster that is *not* sample data

![image](https://user-images.githubusercontent.com/5295965/136103761-47ee29fd-e23e-4b69-82f2-a8ec4eb3716d.png)

To test this:
1. Start Elasticsearch with security disabled: `yarn es snapshot -E xpack.security.enabled=false`
2. Start Kibana 
3. Navigate to Dev Tools and add this staging data:
   ```
   POST foo/_doc
   { "bar": "baz" }
4. Reload the page

You should see the toast now.
You can dismiss the toast, or you can disable the warning in kibana.yml using either `security.showInsecureClusterWarning: false` or `xpack.security.showInsecureClusterWarning: false`.
(I deprecated the former in favor of the latter)

### 2. Share Public URL

When [anonymous access is enabled with another form of authentication](https://www.elastic.co/guide/en/kibana/current/kibana-authentication.html#_anonymous_access_and_other_types_of_authentication), using the Share menu on a supported app (such as Dashboard) should show a switch for the Public URL. Enabling that will add an auth provider hint for the anonymous auth provider.

![image](https://user-images.githubusercontent.com/5295965/136102781-42e37287-c16b-471d-b2e5-6e3c036a0ec8.png)

To test this:
1. Configure Kibana with anonymous access:
   ```yml
   xpack.security.authc.providers:
     basic.basic1:
       order: 0
     anonymous.anonymous1:
       order: 1
       credentials:
         username: "elastic"
         password: "changeme"
   ```
2. Start Elasticsearch and Kibana, and add some sample data
3. View a Dashboard, click Share, and verify that the Public URL switch is displayed

If you configure Kibana without the anonymous auth provider, or with _only_ the anonymous auth provider, the switch should not be displayed.
